packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -135,51 +133,54 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
         }
 
         // fired once the spec bridge is set up and ready to receive messages
-        communicator.once('bridge:ready', () => {
-          // now that the spec bridge is ready, instantiate Cypress with the current app config and environment variables for initial sync when creating the instance
-          communicator.toSpecBridge('initialize:cypress', {
-            config: preprocessConfig(Cypress.config()),
-            env: preprocessEnv(Cypress.env()),
-          })
-
-          state('readyForMultiDomain', true)
-
-          // once the secondary domain page loads, send along the
-          // user-specified callback to run in that domain
-          try {
-            communicator.toSpecBridge('run:domain:fn', {
-              data,
-              fn: callbackFn.toString(),
-              // let the spec bridge version of Cypress know if config read-only values can be overwritten since window.top cannot be accessed in cross-origin iframes
-              // this should only be used for internal testing. Cast to boolean to guarantee serialization
-              // @ts-ignore
-              skipConfigValidation: !!window.top.__cySkipValidateConfig,
-              state: {
-                viewportWidth: Cypress.state('viewportWidth'),
-                viewportHeight: Cypress.state('viewportHeight'),
-                runnable: serializeRunnable(Cypress.state('runnable')),
-                duringUserTestExecution: Cypress.state('duringUserTestExecution'),
-                hookId: state('hookId'),
-              },
+        communicator.once('bridge:ready', (_data, bridgeReadyDomain) => {
+          if (bridgeReadyDomain === domain) {
```

Comment:
We can verify the bridge is ready for the correct domain now, not sure if this is ever not the same or if we should do something more drastic, but it's theoretically possible.